src/core/utils/building/build-entrypoints.ts

```diff
@@ -24,155 +18,18 @@ export async function buildEntrypoints(
   const steps: BuildStepOutput[] = [];
   for (let i = 0; i < groups.length; i++) {
     const group = groups[i];
-
-    spinner.text =
-      pc.dim(`[${i + 1}/${groups.length}]`) +
-      ` ${[group]
-        .flat()
-        .map((e) => e.name)
-        .join(pc.dim(', '))}`;
-
-    const step = Array.isArray(group)
-      ? await buildMultipleEntrypoints(group, config)
-      : await buildSingleEntrypoint(group, config);
-    steps.push(step);
+    const groupNames = [group]
+      .flat()
+      .map((e) => e.name)
+      .join(pc.dim(', '));
+    spinner.text = pc.dim(`[${i + 1}/${groups.length}]`) + ` ${groupNames}`;
+    steps.push(await config.builder.build(group));
   }
   const publicAssets = await copyPublicDirectory(config);
 
-  // Remove any empty directories from moving outputs around
-  await removeEmptyDirs(config.outDir);
-
```

Comment:
This is now handled in the plugin responsible for generating these empty directories.